pkg/monitor/worker.go

```diff
@@ -80,6 +97,21 @@ func (mon *monitor) changefeed(ctx context.Context, baseLog *logrus.Entry, stop
 							fps == api.ProvisioningStateDeleting):
 					mon.deleteDoc(doc)
 				default:
+					// check if we have a Hive shard config and if not start the recursive auth call
+					// in the future we will have the shard index set on the api.OpenShiftClusterDocument
+					// but for now we simply select Hive (AKS) shard 1
+					// e.g. shard := mon.hiveShardConfigs[doc.shardIndex]
+					shard := 1
+
+					mon.shardMutex.RLock()
+					_, exists := mon.hiveShardConfigs[shard]
+					mon.shardMutex.RUnlock()
+					if !exists {
+						// set this to `nil` so cluster monitors will ignore it until its populated with config
+						mon.hiveShardConfigs[shard] = nil
+						go mon.populateHiveShardRestConfig(ctx, shard)
+					}
+
 					// TODO: improve memory usage by storing a subset of doc in mon.docs
 					mon.upsertDoc(doc)
```

Comment:
could we have that in a separate function? the function is already long enough as it is

Reply:
Done.